lib/features/exhibition_flow/screens/voucher_code_screen.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:givt_app_kids/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app_kids/features/exhibition_flow/widgets/voucher_code_input.dart';
+import 'package:givt_app_kids/features/flows/cubit/flows_cubit.dart';
+import 'package:givt_app_kids/features/profiles/cubit/profiles_cubit.dart';
+import 'package:givt_app_kids/features/profiles/models/profile.dart';
+import 'package:givt_app_kids/helpers/snack_bar_helper.dart';
+import 'package:givt_app_kids/shared/widgets/floating_action_button.dart';
+import 'package:go_router/go_router.dart';
+
+import 'package:google_fonts/google_fonts.dart';
+
+class VoucherCodeScreen extends StatefulWidget {
+  const VoucherCodeScreen({super.key});
+
+  @override
+  State<VoucherCodeScreen> createState() => _VoucherCodeScreenState();
+}
+
+class _VoucherCodeScreenState extends State<VoucherCodeScreen> {
+  String _voucherCode = '';
+
+  void _updateVoucherCode(String code) {
+    setState(() {
+      _voucherCode = code;
+    });
+  }
+
+  void _showErrorMessage() {
+    SnackBarHelper.showMessage(
+      context,
+      text: 'Cannot login with voucher. Please try again later.',
+      isError: true,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.sizeOf(context);
+    return BlocConsumer<AuthCubit, AuthState>(
+      listener: (context, authState) {
+        if (authState is ExternalErrorState) {
+          _showErrorMessage();
+        } else if (authState is LoggedInState) {
+          context
+              .read<ProfilesCubit>()
+              .fetchProfiles(authState.session.userGUID);
+        }
+      },
+      builder: (context, authState) {
+        return BlocConsumer<ProfilesCubit, ProfilesState>(
+          listener: (context, propfilesState) {
+            if (propfilesState is ProfilesExternalErrorState) {
+              _showErrorMessage();
+            } else if (propfilesState is ProfilesUpdatedState &&
+                propfilesState.activeProfile == Profile.empty()) {
+              context
+                  .read<ProfilesCubit>()
+                  .setActiveProfile(propfilesState.profiles[0]);
+              context.read<FlowsCubit>().startExhibitionFlow();
+
+              context.pushNamed(Pages.scanNFC.name);
+            }
+          },
+          builder: (context, propfilesState) {
+            return Scaffold(
+              body: authState is LoadingState ||
+                      propfilesState is ProfilesLoadingState
+                  ? const Center(child: CircularProgressIndicator())
+                  : SafeArea(
+                      child: SizedBox.expand(
+                        child: SingleChildScrollView(
+                          child: Column(
+                            mainAxisSize: MainAxisSize.min,
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: [
+                              SizedBox(height: size.height * 0.1),
```

Comment:
In most cases I don't think we should use a height*0.x to define a margin or padding, but use a fixes number. 
For this page it's not really clear for me in the design how we want to have it and its just the voucher screen.. so I guess for now this is fine.

Reply:
also I start playing a bit with FractionallySizedBox, but for me it works a bit buggy with scroll, or idk how to use it yet))